pkg/build/controller/strategy/custom.go

```diff
@@ -70,6 +71,8 @@ func (bs *CustomBuildStrategy) CreateBuildPod(build *buildv1.Build, additionalCA
 		}
 	}
 
+	addProxyEnvVars(defaultProxyConfig, &containerEnv)
```

Comment:
Instead of plumbing this through the pod creation strategy, I'm more inclined to use the build defaults functionality.

When we apply build defaults, we already iterate over all build pod containers and adjust the env vars. See https://github.com/openshift/origin/blob/master/pkg/build/controller/build/defaults/defaults.go#L114

Reply:
@adambkaplan is correct, that is how this should be done.  update the builddefaults config and then let apply builddefaults do the work.

Reply:
updated.  @bparees @adambkaplan ptal
this won't pass until the api bump